### PR TITLE
Input file support

### PR DESCRIPTION
**Pull request by connorpoland**

### CODE REVIEW

File: src/floor/RequestSimulator.java [hunk 1]
```diff
@@ -24,35 +27,73 @@ public RequestSimulator(int f) {
...
+		try (BufferedReader br = new BufferedReader(new FileReader(SimulationVars.inputFile))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split(" ");
 
-		try {
-			Thread.sleep(2000);
-		} catch (InterruptedException e) {
+				//our next message
+				if (Integer.parseInt(parts[1]) == floorNum) {
```
Comment (JackyChiu) [inline src/floor/RequestSimulator.java]: Could you set the `parts[x]` as variables? It was a bit hard to read without referring back to the input file format each time

File: src/floor/RequestSimulator.java [hunk 2]
```diff
@@ -24,35 +27,73 @@ public RequestSimulator(int f) {
...
+					}
+					m.setOriginFloor(floorNum);
+					byte[] schedulerData = Message.serialize(m);
+					DatagramPacket schedulerPacket = new DatagramPacket(schedulerData, schedulerData.length, SimulationVars.schedulerAddress, SimulationVars.schedulerPort);
+					
+					f = new FloorMetaMessage(true);
+					f.setStartingFloor(floorNum);
+					f.setDestinationFloor(Integer.parseInt(parts[3]));
+					byte[] metaData = Message.serialize(f);
+					DatagramPacket metaPacket = new DatagramPacket(metaData, metaData.length, SimulationVars.floorAddresses[floorNum], SimulationVars.floorPorts[floorNum]);
```
Comment (JackyChiu) [inline src/floor/RequestSimulator.java]: Don't you want to send the meta message to the destination floor?

File: src/floor/RequestSimulator.java [hunk 3]
```diff
@@ -24,35 +27,73 @@ public RequestSimulator(int f) {
...
+					
+					//sleep until its time to send
+					try {
+						Thread.sleep(getSleepTime(parts[0]));
+					} catch (InterruptedException e) {
+						e.printStackTrace();
+					}
+					
+					//send the messages
+					System.out.println("hello");
```
Comment (JackyChiu) [inline src/floor/RequestSimulator.java]: Mind deleting this?

File: src/floor/RequestSimulator.java [hunk 4]
```diff
@@ -24,35 +27,76 @@ public RequestSimulator(int f) {
...
+		sleepTime += (Integer.parseInt(parts[2].substring(0, 2))*1000);
+		//add mins 
+		sleepTime += (Integer.parseInt(parts[1])*60000);
+		//add hours
+		sleepTime += (Integer.parseInt(parts[0])*3600000);
+		
+		//time until we reach the time of request
+		sleepTime = sleepTime - time;
+		//when the thread wakes up from this message it will be.. time + time slept
+		time = sleepTime + time;
```
Comment (vikrambombhi) [inline src/floor/RequestSimulator.java]: This will only work if the input file is sorted. idk if it is

File: src/floor/RequestSimulator.java [hunk 5]
```diff
@@ -24,35 +27,76 @@ public RequestSimulator(int f) {
...
+					
+					f = new FloorMetaMessage(true);
+					f.setStartingFloor(floorNum);
+					f.setDestinationFloor(Integer.parseInt(destination));
+					byte[] metaData = Message.serialize(f);
+					DatagramPacket metaPacket = new DatagramPacket(metaData, metaData.length, SimulationVars.floorAddresses[floorNum], SimulationVars.floorPorts[floorNum]);
+					
+					//sleep until its time to send
+					try {
+						Thread.sleep(getSleepTime(timeOfReq));
```
Comment (vikrambombhi) [inline src/floor/RequestSimulator.java]: U shouldn't sleep while reading a buffer. Free the file as soon as possible.

File: src/floor/SimulationVars.java [hunk 6]
```diff
@@ -5,8 +5,9 @@
 public  class SimulationVars {
 	public static int numberOfElevators = 3;
 	public static int numberOfFloors = 3;
-
-	public static int elevatorTravelTime = 500;
+	
+	public static int timeScalar = 10;
```
Comment (vikrambombhi) [inline src/floor/SimulationVars.java]: kinda a future nitpick thing but we should probably make gets and setters for these vars

File: src/floor/RequestSimulator.java (hunk 5 above)
Comment (connorpoland) [reply src/floor/RequestSimulator.java]: In my testing I saw that multiple threads can have their noses in the same file out once so I didn't see this as an issue but I like the idea.

File: src/floor/RequestSimulator.java [hunk 7]
```diff
@@ -24,35 +28,41 @@ public RequestSimulator(int f) {
 		}
 
 		floorNum = f;
+		time = -1;
+		file = new File(SimulationVars.inputFile);
 	}
 
-	@Override
+	@Override
```
Comment (vikrambombhi) [inline src/floor/RequestSimulator.java]: Can u get rid of the trailing white space

**Pull request merged**